pkg/cloudprovider/providers/aws/tags.go

```diff
@@ -227,17 +232,20 @@ func (t *awsTagging) createTags(client EC2, resourceID string, lifecycle Resourc
 // Add additional filters, to match on our tags
 // This lets us run multiple k8s clusters in a single EC2 AZ
 func (t *awsTagging) addFilters(filters []*ec2.Filter) []*ec2.Filter {
+	// if there are no clusterID configured - no filtering by special tag names
+	// should be applied to revert to legacy behaviour.
+	if len(t.ClusterID) == 0 {
+		if len(filters) == 0 {
+			// We can't pass a zero-length Filters to AWS (it's an error)
+			// So if we end up with no filters; just return nil
+			return nil
+		}
+		return filters
+	}
 	// For 1.6, we always recognize the legacy tag, for the 1.5 -> 1.6 upgrade
 	// There are no "or" filters by key, so we look for both the legacy and new key, and then we have to post-filter
 	f := newEc2Filter("tag-key", TagNameKubernetesClusterLegacy, t.clusterTagKey())
 	filters = append(filters, f)
-
-	if len(filters) == 0 {
```

Comment:
Can we leave a comment here saying that if we end up with no filters we have to return nil.

It's a nasty bug that I worry we will re-introduce otherwise...